src/migration/1691766267989-migration.ts

```diff
@@ -0,0 +1,156 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+import 'dotenv/config';
+
+export class Migration1691766267989 implements MigrationInterface {
+    /*
+		Adding in update column triggers and relevant column edits
+	 */
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        const schemaName = process.env.NODE_ENV === 'test' ? 'test' : 'public';
+        // Trigger function
+        await queryRunner.query(`CREATE OR REPLACE FUNCTION ${schemaName}.update_column()   
+		RETURNS TRIGGER AS $$
+		BEGIN
+			NEW.updated_at = now();
+			RETURN NEW;   
+		END;
+		$$ language 'plpgsql';`);
+
+        // Column updates
+        await queryRunner.query(
+            `ALTER TABLE IF EXISTS ${schemaName}.pin_audit_log ALTER COLUMN expired_at TYPE timestamp WITH TIME ZONE;`,
+        );
+        await queryRunner.query(
+            `ALTER TABLE IF EXISTS ${schemaName}.pin_audit_log ALTER COLUMN expired_at SET DEFAULT now();`,
+        );
+        await queryRunner.query(
+            `ALTER TABLE IF EXISTS ${schemaName}.pin_audit_log ALTER COLUMN expired_at SET NOT NULL;`,
+        );
+        await queryRunner.query(
+            `ALTER TABLE IF EXISTS ${schemaName}.pin_audit_log ADD COLUMN IF NOT EXISTS updated_at timestamp WITH time zone;`,
+        );
+        await queryRunner.query(
+            `ALTER TABLE IF EXISTS ${schemaName}.pin_audit_log ADD COLUMN IF NOT EXISTS expired_by_name varchar(75);`,
+        );
+        await queryRunner.query(
+            `ALTER TABLE IF EXISTS ${schemaName}.pin_audit_log ADD COLUMN IF NOT EXISTS expired_by_username varchar(75);`,
+        );
```

Comment:
Need to add in field called action into pin_audit_log as well for PIN History table in FE (https://www.figma.com/file/Qu9ANuVsglGqTnbXCrzhlt/BC-VHERS?type=design&node-id=1148-139167&mode=design&t=a7IRphqVrpXS2Aiu-0).

Additionally, types for expiration_reason may need to be updated (but this still needs to be discussed).

Reply:
Do we need a log entry on creation of PIN as well? Because I see a PIN created in that actions table

Reply:
I'm not 100% but it looks like we do based on the front end designs. If this is the case maybe it makes sense to switch expired_at to created_at?

Reply:
Sounds good. I'll add these changes

Reply:
Actually, I think it would make more sense to have a pin_created_at with the actual pin creation date, and leave expired at to be filled in (nullable)